packages/contracts/src/contracts/current/protocol/AssetProxyDispatcher/proxies/ERC20Proxy.sol

```diff
@@ -0,0 +1,80 @@
+/*
+
+  Copyright 2018 ZeroEx Intl.
+
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+
+*/
+
+pragma solidity ^0.4.21;
+
+import "../IAssetProxy.sol";
+import "../../../utils/LibBytes/LibBytes.sol";
+import "../../../utils/Authorizable/Authorizable.sol";
+import { Token_v1 as ERC20Token } from "../../../../previous/Token/Token_v1.sol";
+
+contract ERC20Proxy is
+    LibBytes,
+    Authorizable,
+    IAssetProxy
+{
+
+    /// @dev Transfers ERC20 tokens. Either succeeds or throws.
+    /// @param assetMetadata ERC20-encoded byte array.
+    /// @param from Address to transfer token from.
+    /// @param to Address to transfer token to.
+    /// @param amount Amount of token to transfer.
+    function transferFrom(
```

Comment:
I'm personally a fan of verbosity in public interfaces where we are able to reduce the effort required for a developer to immediately understand what a method call does (without having to lookup it's implementation and read the comments above it's declaration). The nice thing too is that a developer cannot forget to add this useful comment when re-using the method in a new location.

I would also be against adding error-handling behavior to the function name if I felt this behavior were likely to change in the future, requiring yet another re-name. This seems unlikely here.

I do see now though that this might be a larger naming convention discussion and am happy to table it for now.

Reply:
Lastly, let us remember that Solidity dev tools are still lacking (e.g goto definition, inline method definitions, etc...), so it's much more cumbersome to see a method comment if it's not duplicated above every call-site.